### ISSUE TITLE
Code Suggestions

### ISSUE THREAD

**Issue by klau506** [realxinzhao/gcamfaostat#9]
https://github.com/openjournals/joss-reviews/issues/6388

- [x] When I downloaded the raw data from FAOSTAT according to [this section](https://jgcri.github.io/gcamfaostat/articles/vignette_preparing_data.html#download-faostat-raw-data), a warning appeared
```
> FF_download_RemoteArchive(gcamfaostat_metadata(OnlyReturnDatasetCodeRequired = T))
Starting download dataQCL ...
* Starting download dataPP ...
* Starting download dataPD ...
* Starting download dataTCL ...
* Starting download dataTM ...
* Starting download dataSCL ...
* Starting download dataFBS ...
* Starting download dataFBSH ...
* Starting download dataCB ...
* Starting download dataRFN ...
* Starting download dataRL ...
* Starting download dataFO ...
* Starting download dataOA ...
trying URL 'https://zenodo.org/record/8260225/files/SUA_Crops_Livestock_E_All_Data_(Normalized).zip'
Content type 'application/octet-stream' length 64169556 bytes (61.2 MB)
downloaded 61.2 MB

Download complete.
Warning message:
There was 1 warning in `filter()`.
i In argument: `datasetcode == DATASETCODE`.
Caused by warning in `datasetcode == DATASETCODE`:
! longer object length is not a multiple of shorter object length 
```
and only `SUA_Crops_Livestock_E_All_Data_(Normalized)` was downloaded. It seems that the function does not accept a vector as `DATASETCODE`. Either fix it or document accordingly
- [x] Some warnings and notes when performing `devtools::check()` could be easily removed, such as
```
W  checking dependencies in R code (1.5s)
   Unexported object imported by a ':::' call: 'utils:::format.object_size'
     See the note in ?`:::` about the use of this operator.
     Including base/recommended package(s):
     'utils'

N  checking R code for possible problems (13.4s)
   SUA_bal_adjust: no visible binding for global variable 'Tourist
     consumption'
   driver_drake: no visible binding for '<<-' assignment to
     'xml.XML_SUFFIX'
   Undefined global functions or variables:
     Tourist consumption
```
- [x] #12

**Comment by realxinzhao** [comment]
> It seems that the function (FF_download_RemoteArchive) does not accept a vector as DATASETCODE. Either fix it or document accordingly

Good catch. We fixed the function by adding an assertion to only allow single dataset code input and give a message: "Single dataset allowed; consider using a loop or the function `FF_rawdata_info()` for downloading multiple datasets". The function `FF_download_RemoteArchive` is also corrected accordingly. The corresponding documentation is also updated.

>Consider running the package through the [styler package](https://www.tidyverse.org/blog/2017/12/styler-1.0.0/), with a style of your choice, to ensure clear and consistent code style (e.g. styler:::style_active_pkg())

Great suggestion. We tested this as suggested and the style package works great. However, we realize that there is an active branch, [GCAM_BYU2021](https://github.com/JGCRI/gcamfaostat/tree/GCAM_BYU2021), that will be merged soon (for gcamfaostat v1.1). We will make this full on style updates after that merge to avoid potential conflicts. Thanks!

![Styler](https://github.com/JGCRI/gcamfaostat/assets/47983827/d8fc6acd-e1a8-401f-891b-8a554803ed1c)

**Comment by realxinzhao** [comment]
>  Some warnings and notes when performing devtools::check() could be easily removed

Thanks. We have now cleared all the errors. Warnings are also cleared as much as possible.
There are two remaining but unimportant warnings. We double checked out LICENSE specifications, which should be fine. The `qpdf` warning seems to be R version related and being discussed [here](https://github.com/ropensci/qpdf/issues/17). 

```
> checking DESCRIPTION meta-information ... WARNING
  Invalid license file pointers: LICENSE

> checking R/sysdata.rda ... OK
   WARNING
  'qpdf' is needed for checks on size reduction of PDFs
```
